README.md

```diff
@@ -355,6 +356,38 @@ writable.
 
 Always prefer `ReadonlyArray` over a regular `Array` unless it must be possible to modify the Array.
 
+### React props & TypeScript
+
+`interface` is preferred over `type` when defining the component props. Are a little more powerful
+than the corresponding type declaration, but for a set of react props, it likely doesn't matter.
+
+Also is not necessary to type the `returnType` because is inferred from what you actually return, so
+can be omitted most of the time.
+
+So the best approach and example would be something like this:
+
+```
+interface FancyButtonProps {
+  color: string
+}
+
+const FancyButton = (props: FancyButtonProps) => {
+  ...
+}
+```
+
+or if you're using props destructuring:
```

Comment:
Maybe this should even be the first or only way that's shown here, since IMO this is the preferred way of using props (unless you explicitly need the full props object).

Reply:
Done ✅